src/NuGet.Core/NuGet.Build.Tasks.Pack/Pack.targets

```diff
@@ -237,15 +222,12 @@ Copyright (c) .NET Foundation. All rights reserved.
     <MSBuild
       Projects="@(ValidProjectInputForPackGraph)"
       Targets="_AddMetadataToProjectReference"
-      Properties="%(_MSBuildProjectReferenceExistent.SetConfiguration);
-              %(_MSBuildProjectReferenceExistent.SetPlatform);
-              CustomAfterMicrosoftCommonTargets=$(MSBuildThisFileFullPath);
+      Properties="CustomAfterMicrosoftCommonTargets=$(MSBuildThisFileFullPath);
               BuildProjectReferences=false;
-              TargetFramework=$(TargetFramework);
+              _ParentTargetFramework=$(TargetFramework);
```

Comment:
This is better, but there is one case where you'll miss project references. If the project that is referred to is x-targeted and has project references conditioned on $(TargetFramework). If you want to see those, you need to evaluate with best compatible TFM, which the code I sent in email earlier would do. Given the urgency, I could understand worrying about that separately, but I suggest at least filing a bug to add a test for that case and deal with it. Conditional project references should be rare enough for this not to be high priority.